File: Python Excel Examples/WorkbookApiDemo/protectDocument.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.workbook_api.WorkbookApi(configuration)

name = "ProtectDocument.xlsx"
storage = ""
folder = "ExcelDocument"
protection = spirecloudexcel.models.WorkbookProtectionRequest()
protection.password = "123"
protection.protection_type = "ALL"  # Available type: ALL, READONLY, STRUCTURE, WINDOWS
api.protect_document(name, protection, storage=storage, folder=folder)

File: JavaScript Excel Examples/WorkbooksApiDemo/protectDocument.js
(function (){
    var appId = "your id";
    var appKey = "your key"; 
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudExcel= require('../../src/index');
    var apiClient = new SpirecloudExcel.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudExcel.WorkbookApi(apiClient);

    var name = "Workbook.xlsx";
    var protection = new SpirecloudExcel.WorkbookProtectionRequest();
    protection.password = "123";
    protection.ProtectionType = "STRUCTURE";//Available types: ALL, READONLY, STRUCTURE, WINDOWS
    var opts = {folder : "input"};
    instance.protectDocument(name, protection, opts,
        function (error, data, response) {
            if (error) throw error;
        });
})();